### PR TITLE
CUDA Interop Exercise 

### PR DESCRIPTION
**Pull request by StuartDAdams**
This PR contains a new exercise, as well as some minor corrections to the examples 1 & 2.

### CODE REVIEW

File: exercise-01/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Exercise 01: SYCL interop
+-------------------------------
+
+The in this exercise, you must implement an `interop_task` to let a `SYCL` application call `cuBLAS`.
```
Comment (Ruyk) [inline exercise-01/README.md:4]:
```suggestion
In this exercise, you must implement an `interop_task` to let a `SYCL` application call `cuBLAS`.
```

File: exercise-01/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+add_executable(cuda_sgemv sgemv.cu)
+target_compile_features(cuda_sgemv PUBLIC cxx_std_11)
+set_target_properties(cuda_sgemv PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+set_property(TARGET cuda_sgemv PROPERTY BUILD_RPATH "${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}")
+target_link_libraries(cuda_sgemv CUDA::toolkit CUDA::cublas)
+
+add_executable (sycl_sgemv sycl_sgemv.cpp)
+target_compile_features(sycl_sgemv PUBLIC cxx_std_17)
+target_compile_options(sycl_sgemv PUBLIC ${SYCL_FLAGS})
+target_compile_definitions(sycl_sgemv PUBLIC CUDA_NO_HALF)
```
Comment (Ruyk) [inline exercise-01/CMakeLists.txt:27]: May be worth a comment as to why this is needed

File: exercise-01/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+=========
+
+Two source codes are provided. `sgemv.cu` is the original CUDA code calling
+CUBLAS library to perform the vector/matrix multiplication.
+`sycl_sgemv.cpp` is the unfinished SYCL variant that you must complete.
+Running the `sycl_sgemv.cpp` executable at this stage will result in a runtime error.
+
+Both implementations set up the same input data and expect the same output.
+
+Familiarise yourself with the `host_task` by reading through the SYCL source in example-02.
```
Comment (Ruyk) [inline exercise-01/README.md:47]:
```suggestion
Familiarise yourself with the `interop_task` by reading through the SYCL source in example-02.
```

File: exercise-01/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+Exercise 01: SYCL interop
+-------------------------------
+
+In this exercise, you must implement an `interop_task` to let a `SYCL` application call `cuBLAS`.
```
Comment (rodburns) [inline exercise-01/README.md:4]: must -> will

**Pull request merged**